src/feat/online-feature.h

```diff
@@ -41,6 +41,32 @@ namespace kaldi {
 /// @{
 
 
+/// This class serves as a storage for feature vectors with an option to limit
+/// the memory usage by removing old elements. The deleted frames indices are
+/// "remembered" so that regardless of the MAX_ITEMS setting, the user always
+/// provides the indices as if no deletion was being performed.
+/// This is useful when processing very long recordings which would otherwise
+/// cause the memory to eventually blow up when the features are not being removed.
+class RecyclingVector {
+public:
+  /// By default it does not remove any elements.
+  RecyclingVector(int items_to_hold = -1);
+
+  Vector<BaseFloat> *Retrieve(int index) const;
```

Comment:
Can this return a `const Vector<BaseFloat> &`? The object owns the vectors outright. This way they could not be `delete`d by mistake.

Also, I would maybe call the methods `At` and `PushBack`, so they hark back to `std::vector`.

Reply:
+1 for `At` and `PushBack`. 

If we want to return `const &` here I'd go full way and store the `Vector<BaseFloat>`s by value - but last time I checked move operations were still not supported in Kaldi, and I didn't want to make this PR bigger than necessary. 

I don't want to go half-way because what prevents somebody from `PushBack`ing a null pointer, or deleting the object after he pushes it in here? Would gladly use `unique_ptr`, but didn't see it used in the codebase.

WDYT?

Reply:
Yes, I agree, it should better be symmetric.